reasoner/Reasoner.java

```diff
@@ -82,7 +82,7 @@ private boolean isInfer() {
                                                 Context.Query context) {
         ResourceIterator<Conjunction> conjunctions = iterate(disjunction.conjunctions());
         if (!context.options().parallel()) return conjunctions.flatMap(conj -> iterator(conj, filter));
-        else return produce(conjunctions.flatMap(conj -> producers(conj, filter)).toList());
+        else return produce(conjunctions.flatMap(conj -> producers(conj, filter)).toList()).distinct();
```

Comment:
we wanted to remove this no?

Reply:
Yeah but then the basic test fails! So I put it back.